### PR TITLE
Make all certs 8yr expiry

### PR DESCRIPTION
**Pull request by csibbitt**

### CODE REVIEW

**Comment by csibbitt** [comment]
This is relatively untested, except that I did test the duration string and it did result in a certificate with `.status.renewalTime:  2028-03-07T12:07:12Z`

**Comment by csibbitt** [comment]
Hours is the largest supported unit, BTW https://pkg.go.dev/time#ParseDuration

**Comment by leifmadsen** [comment]
> Hours is the largest supported unit, BTW https://pkg.go.dev/time#ParseDuration

Huh!

File: roles/servicetelemetry/tasks/_local_signing_authority.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@
         name: '{{ ansible_operator_meta.namespace }}-selfsigned'
         namespace: '{{ ansible_operator_meta.namespace }}'
       spec:
+        duration: 70080h
```
Comment (leifmadsen) [inline roles/servicetelemetry/tasks/_local_signing_authority.yml:11]: I wonder if this should be put in as a value read from `defaults/main.yml` just so we have a single spot to edit this in the future?

Comment (csibbitt) [reply roles/servicetelemetry/tasks/_local_signing_authority.yml:11]: Certainly!

**Comment by leifmadsen** [comment]
Suggestion: add a smoketest check that verifies the certificates show up with an expiry date of now+70080h?

**Comment by csibbitt** [comment]
Here is my proof of testing:

Start with a deliberately failing test
```
$ git diff
diff --git a/roles/servicetelemetry/defaults/main.yml b/roles/servicetelemetry/defaults/main.yml
index 2ba6d4b..6db4b5e 100644
--- a/roles/servicetelemetry/defaults/main.yml
+++ b/roles/servicetelemetry/defaults/main.yml
@@ -6,7 +6,7 @@ clouds_remove_on_missing: false
 # default observability strategy (compatible with STF 1.3)
 observability_strategy: use_community
 
-certificate_duration: 70080h
+certificate_duration: 10080h
 
 servicetelemetry_defaults:
   high_availability:
```

Build/deploy
```
$ ansible-playbook  --extra-vars __local_build_enabled=true ./build/run-ci.yaml
[...]
        "* [info] CI Build complete. You can now run tests."
```

Fail
```
$ ./tests/smoketest/smoketest.sh 
[...]
*** [INFO] Checking that the qdr certificate has a long expiry
[FAILURE] Certificate expire time (Jan  2 22:09:16 2024 GMT) less than 7 years from now 
[...]
*** [FAILURE] Smoke test job still not succeeded after 300s

$ echo $?
1
```

Fix
```
$ git reset --hard
HEAD is now at 7435472 Use certificate_duration and test aganist generated cert
```

Build/deploy again...

Pass
```
$ ./tests/smoketest/smoketest.sh 
[...]
*** [INFO] Checking that the qdr certificate has a long expiry
*** [INFO] Waiting to see SNMP trap message in webhook pod      
[...]
*** [SUCCESS] Smoke test job completed successfully

$ echo $?
0
```

**Comment by csibbitt** [comment]
test

**Review by leifmadsen — APPROVED** [review]

**Review by leifmadsen — APPROVED** [review]
Looks great, thanks for adding the test!

**Comment by csibbitt** [comment]
I added a do-not-merge because I want to confirm for sure that going from one cert manifest to the other doesn't mess with your existing certs. I'm 99% sure that you need to manually delete the secret before any changes take effect, but I'd like to get that tested and recorded before merging this.

**Comment by csibbitt** [comment]
> I added a do-not-merge because I want to confirm for sure that going from one cert manifest to the other doesn't mess with your existing certs. I'm 99% sure that you need to manually delete the secret before any changes take effect, but I'd like to get that tested and recorded before merging this.

Right, so I was 99% wrong, it seems. It does mess with the existing cert immediately, but it affects the certificate only, not the keys , so I think the certs should all continue to validate at least until the original expiry of the Issuer's CA cert (that has been copied to OSP). More testing is going to be required to prove that the OSP-side QDR will continue to validate the new cert that is presented. I think it will, but it has me slightly nervous.

**Comment by csibbitt** [comment]
Okay, I think this is safe to merge, here is my testing that shows that updating to this build and getting a new "duration" set doesn't cause any changes in the certificates or secrets themselves.

## Dump all the certs Renewal Time and key/cert hashes
```
$ oc get certificates -o json | jq -r '.items[] | "\(.status.renewalTime) - \(.spec.secretName)"' ; for s in $(oc get secret | grep kubernetes.io/tls | grep -v proxy-tls | cut -d" " -f 1); do oc get secret $s -o json | jq -r .data.\"tls.key\" | sha1sum | head -c 7; echo -n " "; oc get secret $s -o json | jq -r .data.\"tls.crt\" | sha1sum | head -c 7 ; echo " - $s"; done

2023-01-08T21:17:59Z - default-interconnect-inter-router-ca
2023-01-08T21:18:03Z - default-interconnect-inter-router-credentials
2023-01-08T21:17:56Z - default-interconnect-openstack-ca
2023-01-08T21:17:57Z - default-interconnect-openstack-credentials
2023-01-08T21:17:55Z - default-interconnect-selfsigned
2023-01-08T21:18:27Z - elasticsearch-es-cert
2023-01-08T21:18:25Z - service-telemetry-ca
61527d9 59c1c61 - default-interconnect-inter-router-ca
3b8e902 d811d8c - default-interconnect-inter-router-credentials
05713c5 c4fc0e0 - default-interconnect-openstack-ca
f537bee 277fe91 - default-interconnect-openstack-credentials
2fec4a8 af44fb8 - default-interconnect-selfsigned
f75b17c d2470ac - elastic-operator-service-cert
76e78cd 9d87f38 - elasticsearch-es-cert
431a04c 1df1d1a - service-telemetry-ca
```
You can see these are the 90 day certs with a 60 day renewal time

## Deploy new bulid with duration setting
```
$ oc delete csv service-telemetry-operator.v1.5.1668028668 smart-gateway-operator.v5.0.1668028657
$ oc delete buildconfig --all
$ ansible-playbook  --extra-vars __local_build_enabled=true ./build/run-ci.yaml
```

## Verify that duration field has been added
```
$ oc get certificate default-interconnect-selfsigned -o yaml | grep duration
  duration: 70080h0m0s
```

## Verify that actual certificates data and secrets are unchanged
```
oc get certificates -o json | jq -r '.items[] | "\(.status.renewalTime) - \(.spec.secretName)"' ; for s in $(oc get secret | grep kubernetes.io/tls | grep -v proxy-tls | cut -d" " -f 1); do oc get secret $s -o json | jq -r .data.\"tls.key\" | sha1sum | head -c 7; echo -n " "; oc get secret $s -o json | jq -r .data.\"tls.crt\" | sha1sum | head -c 7 ; echo " - $s"; done
2023-01-08T21:17:59Z - default-interconnect-inter-router-ca
2023-01-08T21:18:03Z - default-interconnect-inter-router-credentials
2023-01-08T21:17:56Z - default-interconnect-openstack-ca
2023-01-08T21:17:57Z - default-interconnect-openstack-credentials
2023-01-08T21:17:55Z - default-interconnect-selfsigned
2023-01-08T21:18:27Z - elasticsearch-es-cert
2023-01-08T21:18:25Z - service-telemetry-ca
61527d9 59c1c61 - default-interconnect-inter-router-ca
3b8e902 d811d8c - default-interconnect-inter-router-credentials
05713c5 c4fc0e0 - default-interconnect-openstack-ca
f537bee 277fe91 - default-interconnect-openstack-credentials
2fec4a8 af44fb8 - default-interconnect-selfsigned
f75b17c d2470ac - elastic-operator-service-cert
76e78cd 9d87f38 - elasticsearch-es-cert
431a04c 1df1d1a - service-telemetry-ca
```

I assume that at renewal time we'll get the longer duration, but haven't verified this yet.

**Pull request merged**